migrate/src/main/scala/migrate/ScalacOptionsMigration.scala

```diff
@@ -0,0 +1,364 @@
+      case s"-YdisableFlatCpCaching$_"          => Valid(scalacOption)
+      case "-Ydrop-comments" | "-Ydrop-docs"    => Valid(scalacOption)
+      case s"-Ydump-sbt-inc$_"                  => Valid(scalacOption)
+      case s"-Yerased-terms$_"                  => Valid(scalacOption)
+      case s"-Yexplain-lowlevel$_"              => Valid(scalacOption)
+      case s"-Yexplicit-nulls$_"                => Valid(scalacOption)
+      case s"-Yforce-sbt-phases$_"              => Valid(scalacOption)
+      case s"-Yfrom-tasty-ignore-list$_"        => Valid(scalacOption)
+      case "-Yno-experimental"                  => Valid(scalacOption)
+      case "-Ylegacy-lazy-vals"                 => Valid(scalacOption)
+      case s"-Yindent-colons$_"                 => Valid(scalacOption)
+      case s"-Yinstrument$_"                    => Valid(scalacOption)
+      case s"-Yinstrument-defs$_"               => Valid(scalacOption)
+      case s"-Yno-decode-stacktraces$_"         => Valid(scalacOption)
+      case s"-Yno-deep-subtypes$_"              => Valid(scalacOption)
+      case s"-Yno-double-bindings$_"            => Valid(scalacOption)
+      case s"-Yno-kind-polymorphism$_"          => Valid(scalacOption)
+      case s"-Yno-patmat-opt$_"                 => Valid(scalacOption)
+      case s"-Yplain-printer$_"                 => Valid(scalacOption)
+      case s"-Yprint-debug$_"                   => Valid(scalacOption)
+      case s"-Yprint-debug-owners$_"            => Valid(scalacOption)
+      case "-YprintLevel"                       => Valid(scalacOption)
+      case s"-Yprint-pos$_"                     => Valid(scalacOption)
+      case s"-Yprint-pos-syms$_"                => Valid(scalacOption)
+      case s"-Yprint-syms$_"                    => Valid(scalacOption)
+      case s"-Yrequire-targetName$_"            => Valid(scalacOption)
+      case "-Yrecheck-test"                     => Valid(scalacOption)
+      case "-Ycc-debug"                         => Valid(scalacOption)
+      case "-Ycc-no-abbrev"                     => Valid(scalacOption)
+      case s"-Yretain-trees$_"                  => Valid(scalacOption)
+      case s"-Yscala2-unpickler$_"              => Valid(scalacOption)
+      case s"-Yshow-print-errors$_"             => Valid(scalacOption)
+      case s"-Yshow-suppressed-errors$_"        => Valid(scalacOption)
+      case s"-Yshow-tree-ids$_"                 => Valid(scalacOption)
+      case s"-Yshow-var-bounds$_"               => Valid(scalacOption)
+      case s"-Ytest-pickler$_"                  => Valid(scalacOption)
+      case s"-Yunsound-match-types$_"           => Valid(scalacOption)
+      case "-Xcheck-macros" | "--Xcheck-macros" => Valid(scalacOption)
+      case "-Xsemanticdb" | "-Ysemanticdb"      => Ignored
+      case "-Ykind-projector"                   => Ignored
+      case s"-coverage-out$_"                   => Ignored
+      case s"--coverage-out$_"                  => Ignored
```

Comment:
There four are under the title "Scala 3 only" but that doesn't make sense to me. They probably need their own heading.